### PR TITLE
feat: store config in XDG conform location

### LINKED ISSUE #235: feature: Store config in XDG conform location
Checklist:
* [x] I've searched for similar issues and couldn't find anything matching
* [x] I've discussed this feature in the #k8sgpt slack channel

## Is this feature request related to a problem?
* [ ] Yes
* [x] No

## Describe the solution you'd like
The config file should be stored in an XDG conform location. I would suggest the XDG_CONFIG_HOME folder,
which would be `~/.config` on linux.

There is even a neat golang library which offers standardized locations for mac, windows and linux: 
github.com/adrg/xdg

## Benefits for the project and its users

This would help keeping the users home directory clean and be more inline with other applications installed.

## Potential drawbacks

The config file location changes, so this would be a breaking change. We could still support 
the old location for some time though or offer an automatic migration.

## Additional context

Here is some more information about XDG: https://wiki.archlinux.org/title/XDG_Base_Directory

### PR DESCRIPTION
**Pull request by patrickpichler**
Closes #235 

## 📑 Description

The config file is now located in an folder according to the XDG
specification (`XDG_CONFIG_HOME`).

Since this is a breaking change, there is a check in place, which
informs the end user to please migrate to the new config location, if
the a file at the old config path has been located.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

The config location will change. There is a check in place, which will display a warning to the end user.

### CODE REVIEW

**Comment by matthisholleville** [comment]
Hello, I see that you have changed the name of the configuration file ( old : `.k8sgpt.yaml` / new : `k8sgpt.yaml`). Is this not a breaking change? Have you thought of a way to migrate the data (filters, etc...) from the old configuration to the new one? Maybe just rename the old configuration file if the file exists?

File: cmd/root.go [hunk 1]
```diff
@@ -97,3 +101,37 @@ func initConfig() {
...
+	// ignoring the error here, as we do not want to exit here even on error
+	if err != nil {
+		return
+	}
+
+	oldConfig := filepath.Join(home, ".k8sgpt.yaml")
+
+	if _, err := os.Stat(oldConfig); err == nil {
+		fmt.Fprintln(os.Stderr, color.RedString("Warning: Legacy config file at `%s` detected!", oldConfig))
+		fmt.Fprintln(os.Stderr, color.RedString("Please migrate to new config file location `%s`", getConfigFilePath()))
```
[the remark this answers is not in the document]
Comment (patrickpichler) [reply cmd/root.go]: The warning text has been updated. Please have another look @matthisholleville

[the remark this answers is not in the document]
Comment (matholleville) [reply cmd/root.go]: Great! Shouldn't we make this migration transparent by performing this migration in the `checkForOldConfig` function ?

[the remark this answers is not in the document]
Comment (matthisholleville) [reply cmd/root.go]: Great! Shouldn't we make this migration transparent by performing this migration in the checkForOldConfig function ?

[the remark this answers is not in the document]
Comment (patrickpichler) [reply cmd/root.go]: As said, I personally prefer to manually adopt the config, but I could implement it, if really required. 

I guess the easiest way is to just move the file in the right place. What do you think?

[the remark this answers is not in the document]
Comment (arbreezy) [reply cmd/root.go]: hey @patrickpichler  nice work, I would also suggest to follow the naming [conventions](https://www.conventionalcommits.org/en/v1.0.0/) when introducing a breaking change.

File: cmd/root.go [hunk 2]
```diff
@@ -97,3 +101,38 @@ func initConfig() {
...
+	home, err := os.UserHomeDir()
+
+	// ignoring the error here, as we do not want to exit here even on error
+	if err != nil {
+		return
+	}
+
+	oldConfig := filepath.Join(home, ".k8sgpt.yaml")
+
+	if _, err := os.Stat(oldConfig); err == nil {
```
Comment (arbreezy) [inline cmd/root.go:128]: Just a suggestion for readability, to keep the "happy path" of the function outside of the if statement and possibly return a boolean and an error if old config was found when function is called.
So `checkForOldConfig` is not producing any fmt output and is only checking if config exists.

File: cmd/root.go [hunk 3]
```diff
@@ -97,3 +101,38 @@ func initConfig() {
...
+	}
+}
+
+// tests if a file exists in the legacy config path and warns the user about it
+func checkForOldConfig() {
+	home, err := os.UserHomeDir()
+
+	// ignoring the error here, as we do not want to exit here even on error
+	if err != nil {
+		return
```
Comment (AlexsJones) [inline cmd/root.go:123]: What's going on here?

File: cmd/root.go (hunk 2 above)
Comment (AlexsJones) [reply cmd/root.go:128]: Yes this function is quite hard to read +1

File: cmd/root.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexsJones) [reply cmd/root.go]: Auto migrate please the old config file so this is not a breaking change 
Happy for you to add something like "Updating config file location from X to Y "

**Review by AlexsJones — CHANGES_REQUESTED** [review]

File: cmd/root.go (hunk 2 above)
Comment (patrickpichler) [reply cmd/root.go]: I've replaced this whole function with `performConfigMigrationIfNeeded`. This function will test if the legacy config
exists and will react accordingly.

**Comment by AlexsJones** [comment]
Thanks for updating @patrickpichler I think we need some conflicts sorted and the DCO signed -cheers!

**Comment by AlexsJones** [comment]
Hey @patrickpichler thanks again, is this ready for review?

**Comment by patrickpichler** [comment]
@AlexsJones it is now ready to review :slightly_smiling_face:

**Review by AlexsJones — APPROVED** [review]
Works beautifully, thank you

```
❯ go run main.go filter list
go: downloading github.com/adrg/xdg v0.4.0
Performing config file migration from `/Users/alex/.k8sgpt.yaml` to `/Users/alex/Library/Application Support/k8sgpt/k8sgpt.yaml`
Active:
> StatefulSet
> NetworkPolicy
> Node
> Service
> Ingress
> PersistentVolumeClaim
> CronJob
> ReplicaSet
> Pod
> Deployment
> VulnerabilityReport (integration)
Unused:
> PodDisruptionBudget
> HorizontalPodAutoScaler
```

**Pull request merged**